tests/org.jboss.tools.cdk.ui.bot.test/src/org/jboss/tools/cdk/ui/bot/test/server/editor/CDK3ServerEditorTest.java

```diff
@@ -90,7 +118,85 @@ private void cleanUp() {
 			serversView = null;
 		}
 	}
+	
+	private void checkEditorStateAfterSave(String location, boolean canSave) {
+		((CDK3ServerEditor) editor).getMinishiftBinaryLabel().setText(location);	
+		AbstractWait.sleep(TimePeriod.getCustom(1));
```

Comment:
Very well... To be honest, usage of AbstractWait.sleep (which is not the core, but Thread.sleep called inside) is the only thing, still I am not fully aware how, that with asyncExec call of editor.doSave assures that code works as intended. I will give it more digging to find out how exactly this works. If this piece of code is missing, code reaches into state where test failed due to missing expected exception from WaitUntil inside performSave (editor was saved, is not dirty anymore - which is wrong) and after capturing the screenshot the real saving comes into place and error dialog is thrown! It all makes no sense to me, to be honest!